pkg/controller/controller_ref_manager_test.go

```diff
@@ -136,8 +136,9 @@ func TestClaimPods(t *testing.T) {
 					productionLabelSelector,
 					controllerKind,
 					func() error { return nil }),
-				pods:    []*v1.Pod{newPod("pod1", productionLabel, &controller), newPod("pod2", testLabel, &controller)},
-				claimed: []*v1.Pod{newPod("pod1", productionLabel, &controller)},
+				pods:     []*v1.Pod{newPod("pod1", productionLabel, &controller), newPod("pod2", testLabel, &controller)},
+				claimed:  []*v1.Pod{newPod("pod1", productionLabel, &controller)},
+				released: []*v1.Pod{newPod("pod1", productionLabel, &controller), newPod("pod2", testLabel, &controller)},
```

Comment:
I didn't write this test, but it looks to me like the intention was that this should only list "pod2". However, it also appears that the test never actual checks the `released` field so this is moot.

FakePodControl doesn't seem to have the machinery yet to distinguish which Pods got released, so that could be why the author never implemented that part of the test. The tricky part is that both adoption and release would show up as calls to PatchPod(), and the only way to tell them apart would be to interpret the raw patch data.

An easier alternative would be to at least check the list of Pod names that got patched (either adopted or released). In this case, for example, "pod1" should not be patched at all since it already belongs to the controller. Even that would require updating FakePodControl.PatchPod() since currently it doesn't save the Pod name.

Do you have any interest in fixing the test overall? If so, I'd be happy to review it. But if not, this PR in its current form unfortunately has no effect.